src/gp_engine.hpp

```diff
@@ -47,10 +47,17 @@ class GPEngine {
     branch_lengths_.setConstant(branch_length);
   };
   void SetSBNParameters(EigenVectorXd q) { q_ = std::move(q); };
-  void ResetLogMarginalLikelihood() { log_marginal_likelihood_ = DOUBLE_NEG_INF; }
-  double GetLogMarginalLikelihood() const { return log_marginal_likelihood_; }
+  void ResetLogMarginalLikelihood() {
+    log_marginal_likelihood_.setConstant(DOUBLE_NEG_INF);
+  }
+  double GetLogMarginalLikelihood() const {
+    return (log_marginal_likelihood_.array() * site_pattern_weights_.array()).sum();
+  }
   EigenVectorXd GetBranchLengths() const { return branch_lengths_; };
-  EigenVectorXd GetLogLikelihoods() const { return log_likelihoods_; };
+  EigenVectorXd GetLogLikelihoods() const {
```

Comment:
Thanks for documenting `log_likelihoods_`. Could you please put some documentation about what this function returns? 

I think that it's a vector indexed by GPCSPs, such that entry i stores the marginal log likelihood over all trees that include a GPCSP indexed by i.

Reply:
Also, given that we have a `log_likelihoods_` matrix and this doesn't return this, I might suggest `GetPerGPCSPLogLikelihoods` or something.

Reply:
Done.